mbuild/pattern.py

```diff
@@ -92,7 +102,10 @@ def apply_to_compound(self, guest, guest_port_name='down', host=None,
         assert n_ports >= self.points.shape[0], "Not enough ports for pattern."
         assert_port_exists(guest_port_name, guest)
         box = host.boundingbox
-        pattern = self.points * box.lengths + box.mins
+        if scale:
```

Comment:
This should probably use the `scale()` method to be consistent and so that ports get adjusted appropriately